### PR TITLE
Fix for spec.deployment strategic merge

### PR DESCRIPTION
**Pull request by gazarenkov**
## Which issue(s) does this PR fix or relate to

https://issues.redhat.com/browse/RHIDP-3740

## PR acceptance criteria

- [x] Tests
- [x] Documentation

## How to test changes / Special notes to the reviewer

```
apiVersion: rhdh.redhat.com/v1alpha2
kind: Backstage
metadata:
  name: bs1
spec:
  deployment:
    patch:
      spec:
        template:
          spec:
            volumes:
              - $patch: replace
                name: dynamic-plugins-root
                persistentVolumeClaim:
                  claimName: dynamic-plugins-root
```

### CODE REVIEW

**Comment by kadel** [comment]
to make sure that we don't forget

/cherry-pick 1.3.x

File: integration_tests/rhdh-config_test.go [hunk 1]
```diff
@@ -90,4 +90,44 @@ var _ = When("create default backstage", func() {
...
+
+				if v.Name == "dynamic-plugins-root" {
+					bsvolume = v
+					break
+				}
+			}
+
+			g.Expect(bsvolume).NotTo(BeNil())
+			g.Expect(bsvolume.Ephemeral).To(BeNil())
+			g.Expect(bsvolume.PersistentVolumeClaim).NotTo(BeNil())
```
[the remark this answers is not in the document]
Comment (rm3l) [reply integration_tests/rhdh-config_test.go:128]: Sure, but if for any reason, a bug is introduced and unfortunately not caught in this dependency, you may want to have simple safeguards next time the dependency is updated..
The goal is not to test the merging library per se, but rather to set some level of expectations on the output, depending on the input provided. I find it safer to see a test fail should its input change.
But okay, let's leave it as it is.

File: integration_tests/rhdh-config_test.go [hunk 2]
```diff
@@ -90,4 +90,44 @@ var _ = When("create default backstage", func() {
...
+
+		backstageName := createAndReconcileBackstage(ctx, ns, bs2.Spec, "")
+
+		Eventually(func(g Gomega) {
+			By("creating Deployment ")
+			deploy := &appsv1.Deployment{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: model.DeploymentName(backstageName)}, deploy)
+			g.Expect(err).To(Not(HaveOccurred()))
+
+			var bsvolume corev1.Volume
```
[the remark this answers is not in the document]
Comment (rm3l) [reply integration_tests/rhdh-config_test.go]: As pointed out above, I think the assignment still needs to be fixed ;)
https://github.com/redhat-developer/rhdh-operator/actions/runs/10681012803/job/29603746878?pr=121#step:5:183

**Comment by rm3l** [comment]
@gazarenkov Just curious, why was this PR closed?

**Review by rm3l — APPROVED** [review]

**Pull request merged**